x-pack/plugins/lens/public/shared_components/legend_settings_popover.tsx

```diff
@@ -158,6 +223,63 @@ export const LegendSettingsPopover: React.FunctionComponent<LegendSettingsPopove
         position={position}
         onPositionChange={onPositionChange}
       />
+      <EuiFormRow
+        display="columnCompressedSwitch"
+        label={i18n.translate('xpack.lens.shared.truncateLegend', {
+          defaultMessage: 'Truncate text',
+        })}
+      >
+        <TooltipWrapper
+          tooltipContent={i18n.translate('xpack.lens.shared.legendVisibleTooltip', {
+            defaultMessage: 'Requires legend to be shown',
+          })}
+          condition={mode === 'hide'}
+          position="top"
+          delay="regular"
+          display="block"
+        >
+          <EuiSwitch
+            compressed
+            label={i18n.translate('xpack.lens.shared.truncateLegend', {
+              defaultMessage: 'Truncate text',
+            })}
+            data-test-subj="lens-legend-truncate-switch"
+            showLabel={false}
+            disabled={mode === 'hide'}
+            checked={truncate ?? true}
+            onChange={onTruncateLegendChange}
+          />
+        </TooltipWrapper>
+      </EuiFormRow>
+      <EuiFormRow
+        label={i18n.translate('xpack.lens.shared.maxLinesLabel', {
+          defaultMessage: 'Maximum lines',
+        })}
+        fullWidth
+        display="columnCompressed"
+      >
+        <TooltipWrapper
+          tooltipContent={
+            mode === 'hide'
+              ? i18n.translate('xpack.lens.shared.legendVisibleTooltip', {
+                  defaultMessage: 'Requires legend to be shown',
+                })
+              : i18n.translate('xpack.lens.shared.legendIsTruncated', {
+                  defaultMessage: 'Requires legend to be truncated',
```

Comment:
Nit pick: I am not sure if this message is clear enough. 
<img width="361" alt="Screenshot 2021-08-17 at 11 47 15" src="https://user-images.githubusercontent.com/4283304/129703882-5ecfc624-09a5-4681-b700-0af0ddc9d3c7.png">

I think something like `Requires text to be truncated` is clearer given that the setting has the label `Truncate text`.